base/src/main/scala/com/github/j5ik2o/akka/persistence/s3/base/metrics/MetricsReporter.scala

```diff
@@ -1,16 +1,15 @@
 package com.github.j5ik2o.akka.persistence.s3.base.metrics
 
-import java.util.UUID
 import akka.actor.DynamicAccess
 import com.github.j5ik2o.akka.persistence.s3.base.config.PluginConfig
 import com.github.j5ik2o.akka.persistence.s3.base.exception.PluginException
-import com.github.j5ik2o.akka.persistence.s3.base.model.{ Context, PersistenceId }
+import com.github.j5ik2o.akka.persistence.s3.base.model.Context
 
 import scala.annotation.unused
 import scala.collection.immutable._
 import scala.util.{ Failure, Success }
 
-trait MetricsReporter {
+abstract class MetricsReporter(val pluginConfig: PluginConfig) {
```

Comment:
MetricsReporter/TraceReporter was changed from a trait to an abstruct class so that the implementation class of MetricsReporter/TraceReporter needs to take PluginConfig in its constructor for convenience of initialization process.